### PR TITLE
FindCenterOfMass performace

### LINKED ISSUE #32493: Performance regression in FindCenterOfMassPosition
The algorithm, running in 0.2 seconds on mantid 6.1, now takes over 30 seconds on `release-next` #32350

**To Reproduce**
Download the attached file, chop out `.zip` from the name, and load it.
Run and time the algorithm with BeamRadius=0.1.
The outputs are the same as in 6.1, but it takes 150 times longer now.

This is an extremely critical part of ILL SANS reduction, as it might be run hundreds of times during a complete experiment treatment. After the merge of the PR above, it is now the new bottleneck.
[017112.nxs.zip](https://github.com/mantidproject/mantid/files/7161341/017112.nxs.zip)

### PR DESCRIPTION
**Pull request by peterfpeterson**
There was a performance regression in `FindCenterOfMass` v2 which this addresses. The main cause was getting a handle to `SpectrumInfo` inside of a loop. Other renames and a healthy number of `const` has been added as well.

**To test:**

Try out the original issue and see that it is much better.

Fixes #32493

*This does not require release notes* because it fixes a performance regression introduced during the development cycle.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by gvardany** [comment]
Thanks for reacting swiftly, before I test, could you add a performance test already?

**Review by gvardany — DISMISSED** [review]
Repaired, performance test not urgent as they were not run since last year.

File: Framework/Algorithms/src/FindCenterOfMassPosition2.cpp [hunk 1]
```diff
@@ -114,14 +86,15 @@ double initBoundingBox(WorkspaceBoundingBox &boundingBox, const int numSpec, con
 
 // In subsequent iterations check if spectra fits in the normalized bounding box(generated by previous iterations)
 // If so, update position
-double updateBoundingBox(WorkspaceBoundingBox &boundingBox, WorkspaceBoundingBox previousBoundingBox, const int numSpec,
-                         const double beamRadius, const bool directBeam) {
+double updateBoundingBox(WorkspaceBoundingBox &boundingBox, WorkspaceBoundingBox &previousBoundingBox,
+                         const int numSpec, const double beamRadius, const bool directBeam) {
   double totalCount = 0;
+  const auto spectrumInfo = boundingBox.getWorkspace()->spectrumInfo();
```
Comment (martyngigg) [inline Framework/Algorithms/src/FindCenterOfMassPosition2.cpp:92]: The `ExperimentInfo::spectrumInfo` method returns a `const&` - can we further avoid a copy with `const auto&` here?

Comment (gvardany) [reply Framework/Algorithms/src/FindCenterOfMassPosition2.cpp:92]: so the problem wasn't as much having it in the loop, but copying it in the loop

Comment (martyngigg) [reply Framework/Algorithms/src/FindCenterOfMassPosition2.cpp:92]: Yep. I'm happy to merge this if we want to fix this later? Just while we were here I thought we might as well avoid as much as possible as their could  be some expensive vectors inthe info object.

Comment (gvardany) [reply Framework/Algorithms/src/FindCenterOfMassPosition2.cpp:92]: I'd prefer fixing now, as it's a one character thing.

**Review by martyngigg — APPROVED** [review]

**Review by gvardany — APPROVED** [review]

**Pull request merged**